src/main/java/com/minelittlepony/hdskins/client/gui/GuiSkins.java

```diff
@@ -427,29 +426,37 @@ public void render(int mouseX, int mouseY, float partialTick) {
     }
 
     private void punchServer(String uploadMsg) {
-        uploader.uploadSkin(uploadMsg).handle(CallableFutures.callback(this::updateButtons));
+        uploader.uploadSkin(uploadMsg).whenComplete((o, t) -> {
```

Comment:
Use `CallableFutures.callback`. That was created _specifically_ to avoid this mess.

Reply:
That was more of a hassle to handle. I prefer being explicit.

Reply:
I prefer not looking at a pyramid of completable-future lambdas.